src/mmw/js/src/core/modals/views.js

```diff
@@ -252,6 +254,58 @@ var MultiShareView = ModalBaseView.extend({
     onLinkToggle: function(e) {
         this.model.set('is_private', !e.target.checked);
         this.model.saveProjectAndScenarios();
+    },
+
+    connectHydroShare: function() {
+        var userHasHydroShareAccess = this.options.app.user.get('hydroshare');
+
+        if (userHasHydroShareAccess) {
+            // User already has connected their account. Allow them to turn
+            // this on and off at will.
+            // TODO Add project export ability.
+            return;
+        } else {
+            // User has not connectd to HydroShare yet. Have them sign in
+            // and allow MMW access, then enable the checkbox.
+
+            var self = this,
+                checkbox = self.ui.hydroShareEnabled,
+                confirm = new ConfirmLargeView({
+                        model: new models.ConfirmModel({
+                            titleText: 'Publish on HydroShare',
+                            question: [
+                                'You will need to connect your HydroShare ' +
+                                'account to your Model My Watershed account, ' +
+                                'or create a new HydroShare account.',
+
+                                'Lorem ipsum dolor sit amet, consectetur adipiscing elit. ' +
```

Comment:
Do we know that we will be adding more text to this dialog and the lipsum is acting as a "TODO?"

Reply:
It _is_ a bit of a TODO, and matches the wireframes, but also if we don't have a second paragraph in this modal, the one underneath peeks through, and I thinking hiding it completely is cleaner:

![image](https://user-images.githubusercontent.com/1430060/33841759-9653538a-de66-11e7-9cca-3c4b0ee4f7a5.png)

Hopefully we can come up with some copy that does that. /cc @jfrankl

Reply:
For this Publish on Hydroshare modal, can we treat this as a change in the previous modal as opposed to a modal that sits on top of the previous modal? Visually, I think that will make more sense.